drivers/watchdog/wdt_iwdg_stm32.c

```diff
@@ -17,22 +18,32 @@
 
 #include "wdt_iwdg_stm32.h"
 
-/* Minimal timeout in microseconds. */
-#define IWDG_TIMEOUT_MIN	100
-/* Maximal timeout in microseconds. */
-#define IWDG_TIMEOUT_MAX	26214400
+#define IWDG_PRESCALER_MIN	(4U)
+#define IWDG_PRESCALER_MAX	(256U)
+
+#define IWDG_RELOAD_MIN		(0x0000U)
+#define IWDG_RELOAD_MAX		(0x0FFFU)
+
+/* Minimum timeout in microseconds. */
+#define IWDG_TIMEOUT_MIN	(IWDG_PRESCALER_MIN * (IWDG_RELOAD_MIN + 1U) \
+				 * USEC_PER_SEC / LSI_VALUE)
+
+/* Maximum timeout in microseconds. */
+#define IWDG_TIMEOUT_MAX	((uint64_t)IWDG_PRESCALER_MAX * \
+				 (IWDG_RELOAD_MAX + 1U) * \
+				 USEC_PER_SEC / LSI_VALUE)
```

Comment:
could give here the limit of IWDG_TIMEOUT: from 125 µs to 32.768 s

Reply:
How do you mean? As a comment?

It's not always 32.768 s, as the LSI clock can be 32 kHz (most MCUs), 37 kHz (STM32L0/L1) or 40 kHz (STM32F1/F3).